sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Triggers/BlobTriggerAttributeBindingProvider.cs

```diff
@@ -48,19 +51,26 @@ internal class BlobTriggerAttributeBindingProvider : ITriggerBindingProvider
             _blobsOptions = (blobsOptions ?? throw new ArgumentNullException(nameof(blobsOptions))).Value;
             _exceptionHandler = exceptionHandler ?? throw new ArgumentNullException(nameof(exceptionHandler));
             _blobWrittenWatcherSetter = blobWrittenWatcherSetter ?? throw new ArgumentNullException(nameof(blobWrittenWatcherSetter));
-            _messageEnqueuedWatcherSetter = messageEnqueuedWatcherSetter ?? throw new ArgumentNullException(nameof(messageEnqueuedWatcherSetter));
+            _blobTriggerQueueWriterFactory = blobTriggerQueueWriterFactory ?? throw new ArgumentNullException(nameof(blobTriggerQueueWriterFactory));
             _sharedContextProvider = sharedContextProvider ?? throw new ArgumentNullException(nameof(sharedContextProvider));
             _singletonManager = singletonManager ?? throw new ArgumentNullException(nameof(singletonManager));
 
             _nameResolver = nameResolver;
             _loggerFactory = loggerFactory;
+            _logger = loggerFactory.CreateLogger<BlobTriggerAttributeBindingProvider>();
         }
 
         public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
         {
             ParameterInfo parameter = context.Parameter;
             var blobTriggerAttribute = TypeUtility.GetResolvedAttribute<BlobTriggerAttribute>(context.Parameter);
 
+            if (parameter.ParameterType == typeof(PageBlobClient) && blobTriggerAttribute.Kind == BlobTriggerKind.EventGrid)
+            {
+                _logger.LogError("PageBlobClient is not supported with BlobTriggerKind.EventGrid");
```

Comment:
nit: could you use `nameof` or something like that for BlobTriggerKind?

Reply:
fixed